src/function/gds/single_shortest_paths.cpp

```diff
@@ -81,18 +89,23 @@ class SingleSPPathsEdgeCompute : public EdgeCompute {
         parentListBlock = bfsGraph->addNewBlock();
     }
 
-    bool edgeCompute(nodeID_t boundNodeID, nodeID_t nbrNodeID, relID_t edgeID,
-        bool fwdEdge) override {
-        auto shouldUpdate = frontierPair->pathLengths->getMaskValueFromNextFrontierFixedMask(
-                                nbrNodeID.offset) == PathLengths::UNVISITED;
-        if (shouldUpdate) {
-            if (!parentListBlock->hasSpace()) {
-                parentListBlock = bfsGraph->addNewBlock();
+    void edgeCompute(nodeID_t boundNodeID, std::span<const nodeID_t> nbrNodeIDs,
+        std::span<const relID_t> edgeIDs, SelectionVector& mask, bool isFwd) override {
+        size_t activeCount = 0;
+        mask.forEach([&](auto i) {
+            auto shouldUpdate = frontierPair->pathLengths->getMaskValueFromNextFrontierFixedMask(
+                                    nbrNodeIDs[i].offset) == PathLengths::UNVISITED;
+            if (shouldUpdate) {
+                if (!parentListBlock->hasSpace()) {
+                    parentListBlock = bfsGraph->addNewBlock();
+                }
+                bfsGraph->tryAddSingleParent(frontierPair->curIter.load(std::memory_order_relaxed),
+                    parentListBlock, nbrNodeIDs[i] /* child */, boundNodeID /* parent */,
+                    edgeIDs[i], isFwd);
+                mask.getMutableBuffer()[activeCount++] = i;
```

Comment:
I think we can directly get the edgeCompute() functions to update the next frontiers, since at this point we lost the the benefit of just returning a true/false to delegate the work of setting the frontier to the GDSTask. Can you test how much difference that makes?

Reply:
Copying what I'd said on discord here for the record.

I benchmarked that change on the ldbc graph500-30 dataset using the same query I had been using on the smaller datasets:
- master: average of 43.4 seconds
- the vectorized version: average of 38.2 seconds
- the vectorized version which updates the frontier in edgeCompute: average of 37.4 seconds

The difference is relatively small, and I had to expose stuff from the internals GDSFrontier to the edgeCompute functions to get it to run quickly, so I've left it as-is to keep things simple.